api/controller.py

```diff
@@ -1571,14 +1568,20 @@ def borrow(self, collection_name, identifier_type, identifier, hold_id):
         if isinstance(client, ProblemDetail):
             return client
         if identifier_type and identifier:
-            pools = self.load_licensepools(
-                flask.request.library, identifier_type, identifier
-            )
-            if isinstance(pools, ProblemDetail):
-                return pools
-            pools = [p for p in pools if p.collection == collection]
+            pools = self._db.query(LicensePool).join(LicensePool.collection).join(
+                LicensePool.identifier).filter(
+                    Identifier.type==identifier_type
+                ).filter(
+                    Identifier.identifier==identifier
+                ).filter(
+                    Collection.id==collection.id
+                ).all()
```

Comment:
This is a pretty minor thing, but is it necessary to join against Collection rather than filtering against collection_id.